specification/sections/02-providers.md

```diff
@@ -26,6 +26,20 @@ Providers are the "translator" between the flag evaluation calls made in applica
 provider.getMetadata().getName(); // "my-custom-provider"
 ```
 
+#### Requirement 2.1.2
+
+> The provider **MAY** define an `all flag metadata` field or accessor, containing a `flags` field or accessor which is a collection of flag metadata, identifying all available flag keys in the provider.
+
+```typescript
+provider.getAllFlagMetadata(); // { flags: [{"key": "featureA", "type": "boolean"}, {"key": "featureB", "type": "string"}] }
+```
+
+This operation should not be confused with a bulk evaluation of all flags.
+This is an informative operation available to clients for understanding which flag keys are currently active from a provider.
```

Comment:
The method is planned to be named `getAllFlagMetadata` but the description mentions this returns "currently active" flags only. Is "currently active" defined further somewhere?

Reply:
Good point, worth clarifying this. My personal understanding is that it refers to the enabled feature flags in the provider

Reply:
Thanks for catching that! I know `active` is semantically significant for most vendors. I tried to clarify this by changing it to `available`, which can be interpreted different per provider.

Reply:
@dferber90 I hope it clarifies it. What do you think of this enhancement?